### PR TITLE
Documentation improvement `CHAINERX_ENABLE_{BLAS,LAPACK}`

### PR DESCRIPTION
**Pull request by hvy**
Fixes https://github.com/chainer/chainer/issues/8041.

Also improves related existing documentation.

### CODE REVIEW

**Comment by hvy** [comment]
Changed the documentation to more accurately reflect the current behavior and be less general. PTAL.

File: docs/source/chainerx/install/index.rst [hunk 1]
```diff
@@ -21,7 +21,12 @@ Environment variable        Description
 ``CHAINER_BUILD_CHAINERX``  ``1`` to build the ``chainerx`` package along with ``chainer``. ``0`` to skip. Default is ``0``.
 ``CHAINERX_BUILD_CUDA``     ``1`` to build ``chainerx`` with CUDA support. ``0`` to skip. Default is ``0``.
                             See also :ref:`CUDA support <chainerx-install-cuda-support>` section below.
-``CHAINERX_ENABLE_BLAS``    ``1`` to make BLAS enabled. ``0`` to disabled. Default is ``1``.
+``CHAINERX_ENABLE_BLAS``    ``1`` to enable BLAS, ``0`` to disable it. Default is ``1``.
+                            If BLAS is enabled, it is searched for and used if found. If not found, ChainerX will behave as if BLAS
+                            was disabled and use different native implementations.
```
Comment (Crissman) [inline docs/source/chainerx/install/index.rst]:
```suggestion
                            was disabled and use a native library instead.
```

File: docs/source/chainerx/install/index.rst [hunk 2]
```diff
@@ -21,7 +21,12 @@ Environment variable        Description
...
+``CHAINERX_ENABLE_LAPACK``  ``1`` to enable LAPACK, ``0`` to disable it. Default is ``1``.
+                            If LAPACK is enabled, it is searched for and used if found. If not found, ChainerX will behave as if
+                            LAPACK was disabled and cause runtime errors.
```
Comment (Crissman) [inline docs/source/chainerx/install/index.rst]:
```suggestion
                            LAPACK was disabled and may cause runtime errors.
```

File: docs/source/chainerx/install/index.rst (hunk 1 above)
Comment (hvy) [reply docs/source/chainerx/install/index.rst]: I'm not sure if library is the correct phrase. What it's referring to is these lines https://github.com/chainer/chainer/blob/master/chainerx_cc/chainerx/native/native_device/dot.cc#L178-L227, i.e. naive implementations by us in ChainerX. ChainerX is obviously a library but it's not "naive". 😄

File: docs/source/chainerx/install/index.rst (hunk 2 above)
Comment (hvy) [reply docs/source/chainerx/install/index.rst]: I discussed this PR with @niboshi to be clear about reflecting the actual behavior for now (it always cause runtime errors). As for this case, I'd be fine applying your suggestion but would like to check with him first. What do you say? @niboshi

**Comment by hvy** [comment]
Thanks for the comments! I tried replying to them.

File: docs/source/chainerx/install/index.rst (hunk 2 above)
Comment (niboshi) [reply docs/source/chainerx/install/index.rst]: I think @Crissman's version is better (actually I was going to suggest this exact change).

Comment (hvy) [reply docs/source/chainerx/install/index.rst]: Got it, applied!

**Comment by hvy** [comment]
PTAL.

File: docs/source/chainerx/install/index.rst (hunk 1 above)
Comment (Crissman) [reply docs/source/chainerx/install/index.rst]: Ok, adjusted the description.

Comment (hvy) [reply docs/source/chainerx/install/index.rst]: Thanks, applied!

**Comment by hvy** [comment]
Comments are addressed, PTAL.

**Comment by Crissman** [comment]
imosCI, please test this.

**Pull request merged**